Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1001,6 +1001,43 @@ void HTMLElement::setEnterKeyHint(const AtomString& value)
     setAttributeWithoutSynchronization(enterkeyhintAttr, value);
 }
 
+// https://html.spec.whatwg.org/#dom-hidden
+std::optional<std::variant<bool, double, String>> HTMLElement::hidden() const
+{
+    auto value = attributeWithoutSynchronization(HTMLNames::hiddenAttr);
+    if (document().settings().hiddenUntilFoundEnabled() && equalIgnoringASCIICase(value, "until-found"_s))
+        return "until-found"_s;
+    return !value.isNull();
+}
+
+// https://html.spec.whatwg.org/#dom-hidden
+void HTMLElement::setHidden(const std::optional<std::variant<bool, double, String>> value)
+{
+    if (!value) {
+        removeAttribute(hiddenAttr);
+        return;
+    }
+
+    WTF::switchOn(*value, [&](String string) {
+        if (document().settings().hiddenUntilFoundEnabled() && equalIgnoringASCIICase(string, "until-found"_s))
+            setAttributeWithoutSynchronization(hiddenAttr, "until-found"_s);
+        else if (string.isEmpty() || string.isNull())
```

Comment:
```suggestion
        else if (string.isEmpty())
```
Null strings are also empty strings.